src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java

```diff
@@ -120,58 +136,55 @@ public void testNestedListField() throws Exception {
     InitCodeNode innerList = InitCodeNode.create("0");
     InitCodeNode outerList =
         InitCodeNode.createWithChildren("0", InitCodeLineType.ListInitLine, innerList);
-    InitCodeNode expectedStructure =
+    InitCodeNode outerStructure =
         InitCodeNode.createWithChildren("mylist", InitCodeLineType.ListInitLine, outerList);
+    InitCodeNode expectedStructure = InitCodeNode.newRoot();
+    expectedStructure.mergeChild(outerStructure);
 
-    InitCodeNode actualStructure = FieldStructureParser.parse(fieldSpec);
-    Truth.assertThat(checkEquals(actualStructure, expectedStructure)).isTrue();
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(actualStructure, fieldSpec);
+
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test
   public void testFormattedField() throws Exception {
     String fieldSpec = "name%entity";
 
-    HashMap<String, InitValueConfig> initValueMap = new HashMap<>();
     InitValueConfig initValueConfig = InitValueConfig.create("test-api", null);
-    initValueMap.put("name", initValueConfig);
+    InitCodeNode outerStructure = InitCodeNode.createWithValue("name", initValueConfig);
+
+    InitCodeNode expectedStructure = InitCodeNode.newRoot();
+    expectedStructure.mergeChild(outerStructure);
 
-    InitCodeNode expectedStructure = InitCodeNode.createWithValue("name", initValueConfig);
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(
+        actualStructure, fieldSpec, ImmutableMap.of("name", initValueConfig));
 
-    InitCodeNode actualStructure = FieldStructureParser.parse(fieldSpec, initValueMap);
-    Truth.assertThat(checkEquals(actualStructure, expectedStructure)).isTrue();
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test
   public void testFormattedFieldWithValues() throws Exception {
     List<String> fieldSpecs =
         Arrays.asList("formatted_field%entity1=test1", "formatted_field%entity2=test2");
 
-    HashMap<String, InitValueConfig> initValueMap = new HashMap<>();
     InitValueConfig initValueConfig = InitValueConfig.create("test-api", null);
-    initValueMap.put("formatted_field", initValueConfig);
-
-    HashMap<String, InitValue> expectedCollectionValues = new HashMap<>();
-    expectedCollectionValues.put("entity1", InitValue.createLiteral("test1"));
-    expectedCollectionValues.put("entity2", InitValue.createLiteral("test2"));
 
     InitCodeContext context =
         getContextBuilder()
             .initFieldConfigStrings(fieldSpecs)
-            .initValueConfigMap(initValueMap)
+            .initValueConfigMap(ImmutableMap.of("formatted_field", initValueConfig))
```

Comment:
So we only need to provide the formatted field `formatted_field`, and not the components `entity1`, `entity2` that go into it? I should sit down to understand better how this works. We should document it.

Reply:
I don't understand this either. I don't think this should block us, but I should also investigate.

Reply:
I agree on not blocking, but we should understand the code.